src/svg_surface.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015-2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::ops::Deref;
+use std::path::Path;
+
+extern crate libc;
+use libc::{c_char, c_double};
+
+use ffi;
+use ffi::enums::SurfaceType;
+
+use surface::{Surface, SurfaceExt};
+
+#[cfg(feature = "use_glib")]
+use glib::translate::*;
+
+pub struct SVGSurface(Surface);
+
+extern "C" {
+    pub fn cairo_svg_surface_create (filename: *const c_char,
+                                     width_in_points: c_double,
+                                     height_in_points: c_double) -> *mut ffi::cairo_surface_t;
+}
+
+impl SVGSurface {
+    pub fn from(surface: Surface) -> Result<SVGSurface, Surface> {
+        if surface.get_type() == SurfaceType::Svg {
+            Ok(SVGSurface(surface))
+        } else {
+            Err(surface)
+        }
+    }
+
+    #[doc(hidden)]
+    pub unsafe fn from_raw_full(ptr: *mut ffi::cairo_surface_t) -> SVGSurface {
+        Self::from(Surface::from_raw_full(ptr)).unwrap()
+    }
+
+    pub fn create<T: AsRef<Path>>(filename: T, width: f64, height: f64) -> SVGSurface
+    {
+        unsafe { Self::from_raw_full(cairo_svg_surface_create(filename.as_ref().to_glib_none().0, width, height)) }
```

Comment:
This would require that the Cairo filename convention is the same as GLib's. Are they? That is, arbitrary bytes on UNIX, UTF8 on Windows?

And unfortunately `to_glib_none()` is only available if the cairo bindings are compiled with GLib support

Reply:
The docs say
```
This function always returns a valid pointer, but it will return a pointer to a "nil" surface if an error such as out of memory occurs. You can use cairo_surface_status() to check for this.
```

Should we check for this here and elsewhere and directly return an `Err`?

Reply:
i reverted the changes I made before, i'm not really sure how to address this

Reply:
You'll have to check the documentation or the code. It's kind of important to convert correctly or otherwise the function will fail in certain circumstances.